cli/internal/graph/graph.go

```diff
@@ -86,3 +87,106 @@ func getTaskFromPipeline(pipeline fs.Pipeline, taskID string, taskName string) (
+		mergedTaskDefinition.TaskDependencies = taskDef.TaskDependencies
+		mergedTaskDefinition.Inputs = taskDef.Inputs
+		mergedTaskDefinition.OutputMode = taskDef.OutputMode
+		mergedTaskDefinition.Persistent = taskDef.Persistent
+	}
+
+	return mergedTaskDefinition, nil
+}
+
+func (g *CompleteGraph) getTaskDefinitionChain(pkg *fs.PackageJSON, taskID string, taskName string) ([]fs.TaskDefinition, error) {
+	// Start a list of TaskDefinitions we've found for this TaskID
+	taskDefinitions := []fs.TaskDefinition{}
+
+	// Start in the workspace directory
+	turboJSONPath := turbopath.AbsoluteSystemPath(pkg.Dir).UntypedJoin("turbo.json")
+	_, err := fs.ReadTurboConfig(turboJSONPath)
+
+	// If there is no turbo.json in the workspace directory, we'll use the one in root and carry on.
+	if err != nil {
+		rootTaskDefinition, _ := getTaskFromPipeline(g.Pipeline, taskID, taskName)
+		taskDefinitions = append(taskDefinitions, *rootTaskDefinition)
+		return taskDefinitions, nil
+	}
+
+	// For loop until we `break` manually.
+	// We will reassign `turboJSONPath` inside this loop, so that
+	// every time we iterate, we're starting from a new one.
+	for {
+		turboJSON, err := fs.ReadTurboConfig(turboJSONPath)
+		if err != nil {
+			return nil, err
+		}
+
+		// TODO(mehulkar):
+		// 		getTaskFromPipeline allows searching with a taskID (e.g. `package#task`).
+		// 		But we do not want to allow this, except if we're in the root workspace.
+		taskDefinition, err := getTaskFromPipeline(turboJSON.Pipeline, taskID, taskName)
+		if err != nil {
+			// If there was nothing in the pipeline for this task
+			// We can exit
+			break
+		} else {
+			// Add it into the taskDefinitions
+			taskDefinitions = append(taskDefinitions, *taskDefinition)
+
+			// If this turboJSON doesn't have an extends property, we can stop our for loop here.
+			if len(turboJSON.Extends) == 0 {
+				break
+			}
+
+			// TODO(mehulkar): Enable extending from more than one workspace.
+			// TODO(mehulkar): Enable extending from non-root workspace.
+			if len(turboJSON.Extends) > 1 || turboJSON.Extends[0] != util.RootPkgName {
+				return nil, fmt.Errorf(
+					"You can only extend from the root workspace. \"%s\" extends from %v",
+					pkg.Name,
+					turboJSON.Extends,
+				)
+			}
+
+			// If there's an extends property, walk up to the next one, find the workspace it refers to,
+			// and and assign `directory` to it for the next iteration in this for loop.
+			// Note(mehulkar):
+			//		We are looping through all items in Extends, but as of now,
+			// 		and based on the checks above, we only want to read the first item
+			// 		(and we already know that it's the root workspace).
+			for _, workspaceName := range turboJSON.Extends {
+				workspace, ok := g.WorkspaceInfos[workspaceName]
+				if !ok {
+					// TODO: Should this be a hard error?
+					// A workspace was referenced that doesn't exist or we know nothing about
+					break
```

Comment:
I vote yes, much like importing a non-existant package it should fail eagerly and warn you.